src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java

```diff
@@ -15,7 +16,8 @@
     @Override
     public User login(String login, String password) throws AuthException {
         Optional<User> user = userService.findByLogin(login);
-        if (user.isEmpty() || !password.equals(user.get().getPassword())) {
+        if (!HashUtil.hashPassword(password, user.get().getSalt())
```

Comment:
Why did you remove isEmpty check? is it 100% we will have users in Optional? And maybe it would be better to extract the logic of checking hashes passwords to separate method?

Reply:
Thanks. missed that.